### ISSUE TITLE
Permission popup not showing for PermissionsAndroid

### ISSUE THREAD

**Issue by jake41** [facebook/react-native#24221]
## 🐛 Bug Report
Unable to get permission access via PermissionsAndroid in emulator/real-device. It is always showing permission denied. 

## To Reproduce
Ran exactly the same code as in https://facebook.github.io/react-native/docs/permissionsandroid for ACCESS_FINE_LOCATION permission. 

## Expected Behavior
Ideally a request popup should be shown where the user can allow access or deny location use by app. 

## Code Example
```
if (Platform.OS === "android") {
      console.log("os : ANdroid");
      try {
        const granted = await PermissionsAndroid.request(
          PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
          {
            title: "Facex Location Permission",
            message:
              "App needs access to your location " +
              "so we can show you directions",
            buttonNegative: "Cancel",
            buttonPositive: "OK"
          }
        );
        console.log(granted);
        if (granted === PermissionsAndroid.RESULTS.GRANTED) {
          console.log("Location permission allowed");
        } else {
          console.log("Location permission denied");
        }
      } catch (err) {
        console.warn(err);
      }
    }
```

Added below line in Androidmanifest.xml : 
```
    <!-- Access to Location -->
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```

No popup to ask for location request is coming and in console "Location permission denied" is getting printed all the time. I am always getting never_ask_again as the value of **granted** constant. 

## Environment
```
       buildToolsVersion = "28.0.3"
       minSdkVersion = 18
       compileSdkVersion = 28
       targetSdkVersion = 28
       supportLibVersion = "28.0.0"
```
>  React Native Environment Info:
    System:
      OS: macOS 10.14.3
      CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
      Memory: 895.37 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.8.0 - /usr/local/bin/node
      npm: 6.5.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 26, 27, 28
        Build Tools: 27.0.3, 28.0.3
        System Images: android-26 | Google Play Intel x86 Atom, android-27 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5199772
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: ^16.6.1 => 16.7.0
      react-native: ^0.58.5 => 0.58.6
    npmGlobalPackages:
      react-native-cli: 2.0.1

**Comment by jake41** [comment]
Unable to update to latest version at the moment. Is there any other fix for this ?

**Comment by camilne** [comment]
I am on 0.59.3 and have exactly the same issue with the same code. No pop-up is shown and **never_ask_again** is returned immediately. Running on emulator. React-native version 0.57.8 does not have this issue.

**Comment by german970814** [comment]
Same issue v0.59.4. @camilne did you solve it?

**Comment by camilne** [comment]
@german970814 I downgraded to v0.57.8. It seems to have been introduced in 0.58, and I couldn't find a fix for either 0.58 or 0.59.

**Comment by mart2967** [comment]
This was fixed for me by upgrading to 0.59.9 - I had been experiencing PermissionsAndroid silently returning false for every requested permission in DEV (worked as expected in release), but not after upgrading.

**Comment by fangasvsass** [comment]
0.59.9 same issue ,granted aways return true

**Comment by german970814** [comment]
@jake41 @fangasvsass if you change the emulator it does still with no work? I've changed emulator and now runs well

**Comment by ironhide0396** [comment]
Facing  exact issue , current using react-native 0.59.9 . 
Not working in released apks also . 
Any fix for this ? Or a workaround ?

**Comment by hamzawaleed-crewlogix** [comment]
I was unable to read contacts due to missing **Read_Contacts** permission. Fixed it by adding following permissions
`<uses-permission android:name="android.permission.WRITE_CONTACTS" />`
 `<uses-permission android:name="android.permission.READ_CONTACTS" />`

**Comment by hamzawaleed0102** [comment]
I was unable to read contacts due to missing Read_Contacts permission. Fixed it by adding following permissions
`<uses-permission android:name="android.permission.WRITE_CONTACTS" />`
`<uses-permission android:name="android.permission.READ_CONTACTS" />`

**Comment by LucasHFS** [comment]
I'm with this issue in 0.61.5 React Native version.

**Comment by ironhide0396** [comment]
Fixed it by upgrading to 0.61.5
Issue seemed to be due to diffrent compile and target SDK versions

**Comment by anshikaagarwal** [comment]
I'm also facing the same issue in 0.62.0 react native version even with same versions of compile and target SDK versions.
